packages/flutter_tools/lib/src/base/io.dart

```diff
@@ -259,3 +274,63 @@ class _DefaultProcessInfo implements ProcessInfo {
   @override
   int get maxRss => io.ProcessInfo.maxRss;
 }
+
+/// The return type for listNetworkInterfaces.
+class NetworkInterface implements io.NetworkInterface {
+  NetworkInterface(this._delegate);
+
+  final io.NetworkInterface _delegate;
+
+  @override
+  List<io.InternetAddress> get addresses => _delegate.addresses;
+
+  @override
+  int get index => _delegate.index;
+
+  @override
+  String get name => _delegate.name;
+
+  @override
+  String toString() => "NetworkInterface('$name', $addresses)";
+}
+
+typedef NetworkInterfaceLister = Future<List<NetworkInterface>> Function({
+  bool includeLoopback,
+  bool includeLinkLocal,
+  io.InternetAddressType type,
+});
+
+NetworkInterfaceLister _networkInterfaceListerOverride;
+
+// Tests can set up a non-default network interface lister.
+@visibleForTesting
+void setNetworkInterfaceLister(NetworkInterfaceLister lister) {
+  _networkInterfaceListerOverride = lister;
+}
+
+@visibleForTesting
+void resetNetworkInterfaceLister() {
+  _networkInterfaceListerOverride = null;
+}
+
+Future<List<NetworkInterface>> listNetworkInterfaces({
```

Comment:
bit: Add a brief description and/or point to the dart:io method if relevant

Reply:
Done.